.github/workflows/docs.yml

```diff
@@ -1,15 +1,39 @@
 name: GitHub Pages
 
-on:
+on:  # Always attempt html build on push & PR, but only deploy when there is a new tag.
   push:
     branches:
       - main
+    tags:
+      - v*
   pull_request:
     branches:
       - main
-  workflow_dispatch:
+  workflow_dispatch:  # Allows to manually trigger the job from the web UI.
```

Comment:
You should be able to remove the colon after `workflow_dispatch`.

Reply:
pre-commit fails though, seems like invalid yaml without it:

```
check yaml...............................................................Failed
- hook id: check-yaml
- exit code: 1

while scanning a simple key
  in ".github/workflows/docs.yml", line 12, column 3
could not find expected ':'
  in ".github/workflows/docs.yml", line 14, column 1
```